src/actions/assetsActions.js

```diff
@@ -630,14 +642,43 @@ export const resetSearchAssetsResultAction = () => ({
   type: RESET_ASSETS_SEARCH_RESULT,
 });
 
-export const checkForMissedAssetsAction = (transactionNotifications: Object[]) => {
+const getSupportedTokensAction = (supportedAssets: Asset[], currentAssets: AssetsByAccount, account: Account) => {
+  const accountId = getAccountId(account);
+  const currentAccountAssets = get(currentAssets, accountId, {});
+  const currentAccountAssetsTickers = Object.keys(currentAccountAssets);
+
+  // HACK: Dirty fix for users who removed somehow ETH and PLR from their assets list
+  if (!currentAccountAssetsTickers.includes(ETH)) currentAccountAssetsTickers.push(ETH);
+  if (!currentAccountAssetsTickers.includes(PLR)) currentAccountAssetsTickers.push(PLR);
+
+  const updatedAccountAssets = supportedAssets
+    .filter(asset => currentAccountAssetsTickers.includes(asset.symbol))
+    .reduce((memo, asset) => ({ ...memo, [asset.symbol]: asset }), {});
+  return { id: accountId, ...updatedAccountAssets };
+};
```

Comment:
This function is a bit complex, can we have a test to make sure it's doing what we expect it to do? Also, it doesn't really look like an action, I'd just drop the `Action` suffix.

Reply:
Done :)